### ISSUE TITLE
Fix S3994 FP: Uri can not be specified in Attributes

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#9097]
### Description

S3994 should not raise for properties and parameters in Attributes. Rational: In Attributes `System.Uri` can not be set because of [CS0181 - Attribute constructor parameter has type, which is not a valid attribute parameter type](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0181)  and [CS0655 - not a valid named attribute argument because it is not a valid attribute parameter type](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/parameter-argument-mismatch?f1url=%3FappId%3Droslyn%26k%3Dk(CS0655)#arguments-for-attributes)

```cs
public class MyAttribute : Attribute
{
    public MyAttribute(string uri) // FP S3994: Either change this parameter type to 'System.Uri' or 
                                   // provide an overload which takes a 'System.Uri' parameter.

    { }

    public string Uri {  get; set } // FP S3994: Change this property type to 'System.Uri'.
}
```

### Expected behavior

Do not raise in attributes.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1594](https://sonarsource.atlassian.net/browse/NET-1594)

[NET-1594]: https://sonarsource.atlassian.net/browse/NET-1594?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
